File: 01-Front/Saniteau/Models/Abonnés/EnregistreAbonneRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Saniteau.Compteurs.Contract.Model;

namespace Saniteau.Models.Abonnés
{
    public class EnregistreAbonneRequest
    {
        public int IdAbonne { get; set; }
        public int IdAdresse { get; set; }
        public string Nom { get; set; }
        public string Prenom { get; set; }
        public string Adresse { get; set; }
        public string Ville { get; set; }
        public string CodePostal { get; set; }
        public Tarification Tarification { get; set; }
    }
}

File: 01-Front/Saniteau/client-src/model/abonne/EnregistreAbonneRequest.ts
import { Tarification } from '../../model/abonne/Tarification';


export class EnregistreAbonneRequest {
    public IdAbonne: number;
    public IdAdresse: number;
    public Nom: string;
    public Prenom: string;
    public Adresse: string;
    public Ville: string;
    public CodePostal: string;
    public Tarification: Tarification;

    constructor(idAbonne: number, idAdresse: number, nom: string, prenom: string, adresse: string, ville: string, codePostal: string, tarification: Tarification) {
        this.IdAbonne = idAbonne;
        this.IdAdresse = idAdresse;
        this.Nom = nom;
        this.Prenom = prenom;
        this.Adresse = adresse;
        this.Ville = ville;
        this.CodePostal = codePostal;
        this.Tarification = tarification;
    }
}